src/components/utils/events.ts

```diff
@@ -1,31 +1,34 @@
 import { isEmpty } from '../utils';
 
 /**
- * @class EventDispatcher
+ * Mapped type with subscriptions list
  *
- * Has two important methods:
+ * event name -> array of callbacks
+ */
+type Subscriptions<EventMap> = {
+  [name in keyof EventMap]: Array<(data?: EventMap[name]) => unknown>
```

Comment:
```suggestion
  [Key in keyof EventMap]: Array<(data?: EventMap[Key]) => void>
```

Reply:
Also you could make a type alias for `Listener`:

```
type Listener<Data extends any> = (data?: Data) => void;
```

So you could use it here like `Listener<EventMap[Key]>[]`